tests/reg_tests/test_solver_combos.py

```diff
@@ -0,0 +1,217 @@
+# built-ins
+import unittest
+import os
+import copy
+import numpy
+from parameterized import parameterized_class
+
+# MACH classes
+from adflow import ADFLOW
+from adflow import ADFLOW_C
+
+# import the testing utilities that live a few directories up
+import reg_test_utils as utils
+from reg_default_options import adflowDefOpts
+from reg_aeroproblems import ap_simple_cart_cube
+import reg_test_classes
+
+
+baseDir = os.path.dirname(os.path.abspath(__file__))
+
+# tests for different solver combinations using a simple cartesian block.
+# we are interested in these tests converging without seg-faulting
+# and switching between the solvers correctly. We test
+# Euler + laminar NS + RANS, with smoother, ANK, SANK, CSANK ,NK transition order
+# with turbDADI and turbKSP. So 3 eqn types and 2 turbulence solvers, we have 6 tests
+# for both real and complex cases. These do not cover every possible combination, but
+# represent the most common use cases. Testing turbulence solver options with euler
+# and laminar NS makes sure the solver does not segfault etc.
+
+solver_combo_params = [
+    # Euler Tests
+    {
+        "name": "euler_smoother_ank_sank_csank_nk_turbdadi",
+        "options": {
+            "equationtype": "Euler",
+            # euler fails to get to machine zero in parallel...
+            "L2Convergence": 1e-8,
+            "ankuseturbdadi": True,
+        },
+    },
+
+    {
+        "name": "euler_smoother_ank_sank_csank_nk_turbksp",
+        "options": {
+            "equationtype": "Euler",
+            # euler fails to get to machine zero in parallel...
+            "L2Convergence": 1e-8,
+            "ankuseturbdadi": False,
+        },
+    },
+
+    # laminar NS Tests
+    {
+        "name": "laminar_ns_smoother_ank_sank_csank_nk_turbdadi",
+        "options": {
+            "equationtype": "laminar NS",
+            "ankuseturbdadi": True,
+        },
+    },
+
+    {
+        "name": "laminar_ns_smoother_ank_sank_csank_nk_turbksp",
+        "options": {
+            "equationtype": "laminar NS",
+            "ankuseturbdadi": False,
+        },
+    },
+
+    # RANS tests
+    {
+        "name": "rans_smoother_ank_sank_csank_nk_turbdadi",
+        "options": {
+            "equationtype": "RANS",
+            "ankuseturbdadi": True,
+        },
+    },
+
+    {
+        "name": "rans_smoother_ank_sank_csank_nk_turbksp",
+        "options": {
+            "equationtype": "RANS",
+            "ankuseturbdadi": False,
+        },
+    },
+
+]
+
+# common options dict for both real and complex
+gridFile = os.path.join(baseDir, '../../input_files/cube.cgns')
+commonTestOptions = {
+    'gridfile': gridFile,
+    "writevolumesolution": False,
+    'writesurfacesolution': False,
+    'monitorvariables': ['cpu', 'resrho', 'resturb', 'cd'],
+    "mgcycle": "SG",
+    "mgstartlevel": -1,
```

Comment:
Probably don't need this option

Reply:
were you referring to the mgstartlevel? If so, removed it. By the way, does it matter if the mgcycle's value is `'SG'` or `'sg'`?

Reply:
Yes, I was referring to `mgstartlevel`. 

I realized recently that `'SG'` vs. `'sg'` doesn't matter because the string is passed directly to Fortran and Fortran is case insensitive.

Reply:
neat, I am switching to 'sg' to be consistent. removed the mgstartlevel